### PR TITLE
Feature/93/implement order creation endpoint

### PR DESCRIPTION
**Pull request by ryhaldenys**
## Issues
#93 
## Summary of changes🔥
- Implemented endpoint to place the order.
- Implemented tests.
- Implemented java documentation.

### CODE REVIEW

File: code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java [hunk 1]
```diff
@@ -75,4 +77,20 @@ public ErrorObjectDTO handleBadRequestException(final MethodArgumentTypeMismatch
...
+	public ErrorObjectDTO handleEmptyCartException(final EmptyCartException ex) {
+		log.warn("Cart is empty.", ex);
+		return new ErrorObjectDTO().status(HttpStatus.BAD_REQUEST.value())
+				.title(HttpStatus.BAD_REQUEST.getReasonPhrase()).detail(ex.getMessage());
+	}
+
+	@ExceptionHandler(InsufficientProductQuantityException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+	public ErrorObjectDTO handleInsufficientProductQuantityException(final InsufficientProductQuantityException ex) {
+		log.warn("Cart is empty.", ex);
```
Comment (DenysLiubchenko) [inline code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java]: Wrong log message

File: code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java [hunk 2]
```diff
@@ -75,4 +77,20 @@ public ErrorObjectDTO handleBadRequestException(final MethodArgumentTypeMismatch
...
+	@ExceptionHandler(EmptyCartException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
+	public ErrorObjectDTO handleEmptyCartException(final EmptyCartException ex) {
+		log.warn("Cart is empty.", ex);
+		return new ErrorObjectDTO().status(HttpStatus.BAD_REQUEST.value())
+				.title(HttpStatus.BAD_REQUEST.getReasonPhrase()).detail(ex.getMessage());
+	}
+
+	@ExceptionHandler(InsufficientProductQuantityException.class)
+	@ResponseStatus(value = HttpStatus.BAD_REQUEST)
```
Comment (DenysLiubchenko) [inline code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java]: Maybe better to use different status, for example CONFLICT

File: code/orders-domain/pom.xml [hunk 3]
```diff
@@ -8,5 +8,13 @@
   </parent>
 
   <artifactId>orders-domain</artifactId>
+    <dependencies>
```
Comment (DenysLiubchenko) [inline code/orders-domain/pom.xml]: We cannot validate in our domain. I think the validation part must be on api and only validated data can be passed into the hexagon.

**Review by DenysLiubchenko — CHANGES_REQUESTED** [review]

File: code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java (hunk 1 above)
Comment (ryhaldenys) [reply code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java]: Fixed

File: code/orders-domain/pom.xml (hunk 3 above)
Comment (ryhaldenys) [reply code/orders-domain/pom.xml]: Fixed

File: code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java (hunk 2 above)
Comment (ryhaldenys) [reply code/orders-api-rest/src/main/java/com/academy/orders/apirest/common/ErrorHandler.java]: Improved

**Review by DenysLiubchenko — DISMISSED** [review]

**Review by DenysLiubchenko — APPROVED** [review]

**Pull request merged**